### PR TITLE
zh: snapshot-volume backup and restore

### PR DESCRIPTION
**Pull request by fengou1**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)

volume snapshot backup and restore:
1. add performance introduction
2. add faq
3. remove experimental

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v1.4 (TiDB Operator 1.4 versions)
- [ ] v1.3 (TiDB Operator 1.3 versions)
- [ ] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

File: zh/backup-restore-faq.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+现象：BR 数据恢复子任务失败，第一个 BR 恢复子任务恢复成功 （volume complete)，第二个子任务失败。打印失败任务 log 发现以下 log 信息：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    error="rpc error: code = Unavailable desc = keepalive watchdog timeout"
+    ```
+
+问题发生的场景：数据规模较大且使用的 TiDB Cluster v6.3.0 的版本
```
Comment (IANTHEREAL) [inline zh/backup-restore-faq.md:94]: 这个原因是什么？

File: zh/backup-restore-snapshot-perf.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+本节介绍影响备份性能的因素以及性能测试结果。
+以下测试指标基于 AWS region：us-west-2。
+
+## EBS 卷快照备份各阶段的耗时
+
+EBS 卷快照备份阶段包含创建备份任务，停止调度，停止 GC，获取备份时间 backupts 以及 卷快照。其中最耗时的部分是创建卷快照。备份过程中卷的快照创建是并行的，一个备份完成时间取决于用时最长的快照完成时间。
+
+## EBS 卷快照备份的性能
+
+卷快照备份时间取决于最长数据卷快照完成时间，这部分工作由 AWS EBS 服务来完成，当前 AWS 并没有提供卷快照完成量化指标。根据我们的测试，在 TiDB-Operator 推荐配置下 （GP3）整个备份时间大致如下：
```
Comment (IANTHEREAL) [inline zh/backup-restore-snapshot-perf.md:17]: 提示一下 EBS 会采用增量备份。

File: zh/backup-restore-snapshot-perf.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+5. 启动 TiDB：启动 TiDB，恢复完成
+
+### 各阶段耗时总占比
+
+| 恢复阶段     | 恢复大致耗时 | 恢复总占比 | 备注                                          |
+| :--------: | :---------: | :------: | :-------------------------------------------: |
+| 创建集群     | 30s         | %1       | 包含下载 docker image 和启动 pd                 |
+| 卷恢复      | 20s         | %1       | 包含启动 br pod 和卷恢复                         |
+| TiKV 启动   | 3min/30min  | 58%      | 有写入的备份恢复，rocksdb 启动产生额外耗时 2～30min |
```
Comment (IANTHEREAL) [inline zh/backup-restore-snapshot-perf.md:60]: 3 min 和 30 min 分别指什么？

Comment (IANTHEREAL) [reply zh/backup-restore-snapshot-perf.md:60]: ”有写入的备份恢复“  这句话不是很理解

File: zh/backup-restore-snapshot-perf.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+| 数据恢复阶段 | 20min       | 38%      | 20 min 是已验证最大数据量 20 TB 的耗时            |
```
Comment (IANTHEREAL) [inline zh/backup-restore-snapshot-perf.md:61]: 这句话的解释，跟其他的貌似不一样。而且没太明白是啥意思

File: zh/backup-restore-faq.md (hunk 1 above)
Comment (fengou1) [reply zh/backup-restore-faq.md:94]: 6.3 GA 前使用 compaction, 大数据量 TLS 链接会超时。因为有 workaround 也没有去，且 6.4 会使用 flashback, 就没有去更改这块。

File: zh/backup-restore-snapshot-perf.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+本节介绍影响备份的性能、影响因素以及性能测试结果。
+
+### 备份耗时
+
+EBS 卷快照备份阶段包含创建备份任务、停止调度、停止 GC、获取备份时间 backupts 以及卷快照。详细信息，参考[基于 EBS 卷快照的备份恢复功能架构](backup-restore-overview.md)。其中，耗时最多的阶段是创建卷快照。备份过程中卷的快照并行创建，一个备份完成时间取决于用时最后一个卷的快照完成时间。
+
+### 备份耗时占比
+
+| 恢复阶段     | 恢复耗时 | 恢复总占比 | 备注                                     |
```
Comment (shichun-0415) [inline zh/backup-restore-snapshot-perf.md:20]: 这个表的数据应该都是备份的？

Comment (fengou1) [reply zh/backup-restore-snapshot-perf.md:20]: 是的

**Review by XuJianxu — APPROVED** [review]
LGTM

File: zh/backup-restore-faq.md (hunk 1 above)
Comment (fengou1) [reply zh/backup-restore-faq.md:99]: fix

File: zh/backup-restore-snapshot-perf.md (hunk 2 above)
Comment (fengou1) [reply zh/backup-restore-snapshot-perf.md]: fix

File: zh/backup-restore-snapshot-perf.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+    启动 TiDB，恢复完成。
+
+### 恢复耗时占比
+
+| 恢复阶段     | 恢复大致耗时 | 恢复总占比 | 备注                                                            |
+| :--------: | :---------: | :------: | :-------------------------------------------------------------: |
+| 创建集群     | 30 s         | %1       | 包含下载 docker image 和启动 PD                                   |
+| 卷恢复      | 20 s         | %1       | 包含启动 br pod 和卷恢复                                           |
+| 启动 TiKV   | 2～30 min    | 58%      | 备份时有正在进行的事务，此备份做恢复时，rocksdb 启动产生额外耗时 5～30 min |
```
Comment (IANTHEREAL) [inline zh/backup-restore-snapshot-perf.md:85]: 是否在 2 ~ 30 min基础上再加 5 ～ 30 min 吗？

File: zh/backup-restore-snapshot-perf.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+summary: 了解 EBS 卷快照备份恢复的性能基线。
+---
+
+# 基于 EBS 卷快照备份恢复的性能介绍
+
+本文介绍 EBS备份恢复的性能、性能的影响因素以及性能测试结果。以下性能指标基于 AWS region `us-west-2`。
+
+## 备份性能
+
+本节介绍影响备份的性能、影响因素以及性能测试结果。
```
Comment (IANTHEREAL) [inline zh/backup-restore-snapshot-perf.md:12]:
```suggestion
本节介绍影响备份的性能、影响因素以及性能测试结果。 备份性能是可扩展的，随着 TiKV 节点数增加而线性提升。
```

File: zh/backup-restore-snapshot-perf.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+### 备份耗时占比
+
+| 备份阶段     | 备份耗时    | 备份总占比 | 备注                                     |
+| :--------: | :---------: | :------: | :-------------------------------------: |
+| 卷快照时间   | 16 m (50GB) | %99      | AWS EBS snapshot 的时间                  |
+| 其他        | 1s          | %1       | 包含停止调度，停止 GC，获取备份时间 backupts |
+
+### 备份性能数据
+
+卷快照备份时间取决于最后一个数据卷快照完成时间，这部分工作由 AWS EBS 服务来完成，当前 AWS 没有提供卷快照完成量化指标。根据测试，在 TiDB-Operator 推荐机型配置下，使用存储卷类型 gp3，配置 400 MiB/s 与 7000 IOPS，整个备份过程耗时大致如下：
```
Comment (IANTHEREAL) [inline zh/backup-restore-snapshot-perf.md:27]:
```suggestion
卷快照备份的整体时间取决于最后一个数据卷快照完成时间，这部分工作由 AWS EBS 服务来完成，当前 AWS 没有提供卷快照完成量化指标。根据测试，在 TiDB-Operator 推荐机型配置下，使用存储卷类型 gp3，配置 400 MiB/s 与 7000 IOPS，整个备份过程耗时大致如下：
```

File: zh/backup-restore-snapshot-perf.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+### 备份影响
+
+因 AWS EBS 卷快照特点，使用 gp3 卷进行备份时，经过测试集群影响小于 3%。如下图所示，10:25 分之后发起备份。
+
+![EBS Snapshot backup impact](/media/volume-snapshot-backup-impact.jpg)
+
+## 恢复性能
+
+本节介绍恢复的性能、影响因素以及性能测试结果。
```
Comment (IANTHEREAL) [inline zh/backup-restore-snapshot-perf.md:53]:
```suggestion
本节介绍恢复的性能、影响因素以及性能测试结果。恢复性能是可扩展的，随着 TiKV 节点数的增加而提升
```

File: zh/backup-restore-snapshot-perf.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+| 卷快照时间   | 16 m (50GB) | %99      | AWS EBS snapshot 的时间                  |
+| 其他        | 1s          | %1       | 包含停止调度，停止 GC，获取备份时间 backupts |
+
+### 备份性能数据
+
+卷快照备份时间取决于最后一个数据卷快照完成时间，这部分工作由 AWS EBS 服务来完成，当前 AWS 没有提供卷快照完成量化指标。根据测试，在 TiDB-Operator 推荐机型配置下，使用存储卷类型 gp3，配置 400 MiB/s 与 7000 IOPS，整个备份过程耗时大致如下：
+
+![EBS Snapshot backup perf](/media/volume-snapshot-backup-perf.png)
+
+| 单卷数据  | 卷大小   | 卷配置             | 大概备份时间 |
```
Comment (shichun-0415) [inline zh/backup-restore-snapshot-perf.md:31]: `单卷数据`和`卷大小`看起来概念比较像。

File: zh/backup-restore-snapshot-perf.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+
+> **注意：**
+>
+> 上述性能数据是基于全量数据备份而进行的备份。
+>
+> 因为 AWS 卷快照是以卷为单位，除第一个卷快照是全量，后续卷快照都是以增量的形式进行。每日备份一般情况下可以在 1 小时完成，如遇特殊情况，可缩短备份频率，如 12 小时或者 8 小时备份一次。
+
+### 备份影响
+
+因 AWS EBS 卷快照特点，使用 gp3 卷进行备份时，经过测试集群影响小于 3%。如下图所示，10:25 分之后发起备份。
```
Comment (shichun-0415) [inline zh/backup-restore-snapshot-perf.md:47]:
```suggestion
使用 gp3 卷进行备份时，经过测试集群影响小于 3%。如下图所示，10:25 分之后发起备份。
```
因 AWS EBS 卷快照特点，这句在这里没有实际意义。

File: zh/backup-restore-snapshot-perf.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+| :--------: | :---------: | :------: | :-------------------------------------------------------------: |
+| 创建集群     | 30 s         | %1       | 包含下载 docker image 和启动 PD                                   |
+| 卷恢复      | 20 s         | %1       | 包含启动 br pod 和卷恢复                                           |
+| 启动 TiKV   | 2～30 min    | 58%      | 备份时有正在进行的事务，此备份做恢复时，rocksdb 启动产生额外耗时 5～30 min |
+| 数据恢复阶段 | 2～20 min    | 38%       | 包含 raft 共识层数据恢复以及 MVCC 删除数据操作                        |
+| 启动 TiDB   | 1 min        | 2%       | 包含下载 tidb docker image 和启动 TiDB                          |
+
+> **注意：**
+>
+> 因为卷快照是 crash consistency 的状态，EBS 卷快照恢复阶段 TiKV 启动阶段，需要先启动 kv db (rocksdb)，此阶段在有写入的备份恢复中，会产生额外的耗时。经测试，额外的耗时在 30 分钟以内，如使用高性能盘恢复，如 io2, 额外耗时可缩短到 5 分钟内，如果使用标准 gp3 盘 （3000IOPS/125MiB/s），额外耗时可能到达到 30 分钟左右。
```
Comment (shichun-0415) [inline zh/backup-restore-snapshot-perf.md:91]: 1. `快照是 crash consistency 的状态`，请转换成更容易理解的表达。
2. 额外消耗时间一会儿描述是 30 分钟内，一会儿是 5 分钟以内，一会儿又是 30 分钟左右，读着比较晕，建议调整，提供最关键的信息。
3. `EBS 卷快照恢复阶段 TiKV 启动阶段，需要先启动 kv db (rocksdb)，此阶段在有写入的备份恢复中`，这句话里涉及好几个 `阶段`，描述不清晰，需要调整。

File: zh/backup-restore-snapshot-perf.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+| 启动 TiDB   | 1 min        | 2%       | 包含下载 tidb docker image 和启动 TiDB                                           |
+
+> **注意：**
+>
+> 因为卷快照是 crash consistency 的状态，EBS 卷快照恢复阶段 TiKV 启动阶段，需要先启动 kv db (rocksdb)，此阶段在有写入的备份恢复中，会产生额外的耗时。经测试，额外的耗时在 30 分钟以内，如使用高性能盘恢复，如 io2, 额外耗时可缩短到 5 分钟内，如果使用标准 gp3 盘 （3000IOPS/125MiB/s），额外耗时可能到达到 30 分钟左右。
+>
+
+### 恢复性能数据
+
+卷快照恢复时间取主要决于 TiKV 启动和数据恢复阶段，TiKV 启动受 EBS 卷快照恢复延迟初始化影响，启动时间在2分钟到30分钟之间，这部分工作由 AWS EBS 服务来完成，当前 AWS 并没有提供卷快照完成量化指标。根据测试，在 TiDB-Operator 推荐 EC2 机型配置下 使用 gp3 如下配置，整个恢复时间大致如下：
```
Comment (shichun-0415) [inline zh/backup-restore-snapshot-perf.md:96]: `这部分工作由 AWS EBS 服务来完成`，这部分指哪部分？前面提到了启动 TiKV 和数据恢复阶段。

File: zh/backup-restore-snapshot-perf.md (hunk 6 above)
Comment (fengou1) [reply zh/backup-restore-snapshot-perf.md]: rephrase in https://github.com/pingcap/docs-tidb-operator/pull/2108/commits/61f68c983edf91701ef66a22f1cd27b7389b4e17

File: zh/backup-restore-snapshot-perf.md (hunk 13 above)
Comment (fengou1) [reply zh/backup-restore-snapshot-perf.md]: fix in https://github.com/pingcap/docs-tidb-operator/pull/2108/commits/b5a5387cad19af617ad1fcde563bf93990b4fb0d

File: zh/backup-restore-snapshot-perf.md (hunk 12 above)
Comment (fengou1) [reply zh/backup-restore-snapshot-perf.md]: fix in https://github.com/pingcap/docs-tidb-operator/pull/2108/commits/b5a5387cad19af617ad1fcde563bf93990b4fb0d

File: zh/backup-restore-snapshot-perf.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+
+### 备份耗时
+
+EBS 卷快照备份阶段包含创建备份任务、停止调度、停止 GC、获取备份时间 backupts 以及卷快照。详细信息，参考[基于 EBS 卷快照的备份恢复功能架构](backup-restore-overview.md)。其中，耗时最多的阶段是创建卷快照。备份过程中卷的快照并行创建，一个备份完成时间取决于用时最后一个卷的快照完成时间。
+
+### 备份耗时占比
+
+| 备份阶段     | 备份耗时    | 备份总占比 | 备注                                     |
+| :--------: | :---------: | :------: | :-------------------------------------: |
+| 卷快照时间   | 16 m (50GB) | %99      | AWS EBS snapshot 的时间                  |
```
[the remark this answers is not in the document]
Comment (fengou1) [reply zh/backup-restore-snapshot-perf.md]: fix in https://github.com/pingcap/docs-tidb-operator/pull/2108/commits/b5a5387cad19af617ad1fcde563bf93990b4fb0d

File: zh/backup-restore-snapshot-perf.md [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+| :------: | :-----: | :---------------: | :--------: |
+| 50 GB    | 500 GB  | 7000IOPS/400MiB/s | 20 min    |
+| 100 GB   | 500 GB  | 7000IOPS/400MiB/s | 50 min    |
+| 200 GB   | 500 GB  | 7000IOPS/400MiB/s | 100 min   |
+| 500 GB   | 1024 GB | 7000IOPS/400MiB/s | 150 min   |
+| 1024 GB  | 3500 GB | 7000IOPS/400MiB/s | 350 min   |
+
+> **注意：**
+>
+> 上述性能数据是基于全量数据备份而进行的备份。
```
[the remark this answers is not in the document]
Comment (fengou1) [reply zh/backup-restore-snapshot-perf.md]: fix in https://github.com/pingcap/docs-tidb-operator/pull/2108/commits/b5a5387cad19af617ad1fcde563bf93990b4fb0d

File: zh/backup-restore-snapshot-perf.md [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+
+本文介绍 EBS备份恢复的性能、性能的影响因素以及性能测试结果。以下性能指标基于 AWS region `us-west-2`。
+
+## 备份性能
+
+本节介绍影响备份的性能、影响因素以及性能测试结果。
+
+### 备份耗时
+
+EBS 卷快照备份阶段包含创建备份任务、停止调度、停止 GC、获取备份时间 backupts 以及卷快照。详细信息，参考[基于 EBS 卷快照的备份恢复功能架构](backup-restore-overview.md)。其中，耗时最多的阶段是创建卷快照。备份过程中卷的快照并行创建，一个备份完成时间取决于用时最后一个卷的快照完成时间。
```
[the remark this answers is not in the document]
Comment (fengou1) [reply zh/backup-restore-snapshot-perf.md]: fix in https://github.com/pingcap/docs-tidb-operator/pull/2108/commits/b5a5387cad19af617ad1fcde563bf93990b4fb0d

File: zh/backup-restore-faq.md [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+
+issue 链接：[#4778](https://github.com/pingcap/tidb-operator/issues/4778)
+
+现象：删除备份 CR 卡住。
+
+问题发生的场景：使用了 TiDB Operator 版本 v1.4.0-beta.2 或者更早的版本。
+
+解决方案：升级 TiDB Operator 到最新版本。
+
+### 备份发起后立即失败
```
[the remark this answers is not in the document]
Comment (fengou1) [reply zh/backup-restore-faq.md]: fix in https://github.com/pingcap/docs-tidb-operator/pull/2108/commits/b5a5387cad19af617ad1fcde563bf93990b4fb0d

**Review by IANTHEREAL — APPROVED** [review]

File: zh/backup-restore-snapshot-perf.md (hunk 7 above)
Comment (ran-huang) [inline zh/backup-restore-snapshot-perf.md:12]: Backup performance is scalable and increases linearly with the number of TiKV nodes.
中文少了这一句？

Comment (shichun-0415) [reply zh/backup-restore-snapshot-perf.md:16]: 英文删掉了这句。

File: zh/backup-restore-faq.md [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+        kubectl get po -n ${namespace}
+        ```
+
+    * 查看 TiDB Operator controller manager pod 的 log 信息：
+
+        ```shell
+        kubectl -n ${namespace} logs ${tidb-controller-manager}
+        ```
+
+    * 检查 log 信息是否包含以下错误：
```
[the remark this answers is not in the document]
Comment (shichun-0415) [reply zh/backup-restore-faq.md]: Fixed in [c5d7fae](https://github.com/pingcap/docs-tidb-operator/pull/2108/commits/c5d7fae71e39a0b39ec33271f543fc5e2d387f86)

**Review by ran-huang — APPROVED** [review]

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: zh/backup-restore-snapshot-perf.md [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+| 创建集群     | 30s         | 1%      | 包含下载 docker image 和启动 PD 的时间                                                 |
+| 卷恢复      | 20s         | 1%     | 包含启动 BR pod 和卷恢复的时间                                                         |
+| 启动 TiKV   | 2～30 min    | 58%      | TiKV 启动时长受卷性能影响，一般 3 分钟左右，当恢复数据包含事务时，此阶段耗时可能达到 30 分钟 |
+| 数据恢复阶段 | 2～20 min    | 38%       | 包含恢复 raft 共识层数据以及删除 MVCC 数据的时间                                      |
+| 启动 TiDB   | 1 min        | 2%       | 包含下载 docker image 和启动 TiDB 的时间                                           |
+
+> **注意：**
+>
+> 因为卷快照具有崩溃一致性 (Crash Consistency) 的特点，EBS 卷快照恢复阶段 TiKV 启动阶段，需要数据自检和自动修复。经测试，该阶段耗时在 30 分钟以内，如使用高性能盘恢复，如提升 IOPS 和带宽, 额外耗时可缩短到 5 分钟内。使用高性能盘恢复请参考[恢复时间太长（大于 2 小时）](backup-restore-faq.md#恢复时间太长大于-2-小时)。
+>
```
Comment (shichun-0415) [inline zh/backup-restore-snapshot-perf.md:96]:
```suggestion
```

**Pull request merged**